app/controllers/async_info_controller.rb

```diff
@@ -53,25 +53,27 @@ def broadcast_data
   def user_data
     Rails.cache.fetch(user_cache_key, expires_in: 15.minutes) do
       {
-        id: @user.id,
-        name: @user.name,
-        username: @user.username,
-        profile_image_90: ProfileImage.new(@user).get(width: 90),
-        followed_tags: @user.cached_followed_tags.to_json(only: %i[id name bg_color_hex text_color_hex hotness_score], methods: [:points]),
-        followed_user_ids: @user.cached_following_users_ids,
-        followed_podcast_ids: @user.cached_following_podcasts_ids,
-        reading_list_ids: ReadingList.new(@user).cached_ids_of_articles,
         blocked_user_ids: @user.all_blocking.pluck(:blocked_id),
-        saw_onboarding: @user.saw_onboarding,
         checked_code_of_conduct: @user.checked_code_of_conduct,
         checked_terms_and_conditions: @user.checked_terms_and_conditions,
-        display_sponsors: @user.display_sponsors,
+        config_body_class: @user.config_body_class,
+        created_at: @user.created_at,
         display_announcements: @user.display_announcements,
-        trusted: @user.trusted,
+        display_sponsors: @user.display_sponsors,
+        email: @user.email,
+        followed_podcast_ids: @user.cached_following_podcasts_ids,
+        followed_tags: @user.cached_followed_tags.to_json(only: %i[id name bg_color_hex text_color_hex hotness_score], methods: [:points]),
+        followed_user_ids: @user.cached_following_users_ids,
+        id: @user.id,
         moderator_for_tags: @user.moderator_for_tags,
-        config_body_class: @user.config_body_class,
+        name: @user.name,
         pro: @user.pro?,
-        created_at: @user.created_at
+        profile_image_90: ProfileImage.new(@user).get(width: 90),
+        reading_list_ids: ReadingList.new(@user).cached_ids_of_articles,
+        saw_onboarding: @user.saw_onboarding,
+        subscription_source_article_ids: @user.cached_subscription_source_article_ids,
```

Comment:
I alphabetized this list and added:
```ruby
email: @user.email
subscription_source_article_ids: @user.cached_subscription_source_article_ids
```
We want the `email` to populate text in the liquid tag. Something like:
> Hey, user! Are you sure you want to share your email address (test@test.com) with the author_username?

The exact wording is TBD, but we'll want the email address so it's as clear as possible to the user what they're agreeing to.

And we want `subscription_source_article_ids` so we know which Authors the current user is subscribed to and through which articles. Then we know to display the "already subscribed" UX.

Reply:
I see a privacy issue here as this means that _all_ users will have their emai suddenly exposed in the async data info blob

Can we share this in the async blob ONLY if the requesting user is an admin? AFAIK we said only privileged users would be using this feature.

cc @mstruve @vaidehijoshi

Reply:
Couple things I am thinking of looking at this. 
1) `subscription_source_article_ids`: We are going to make a request to Redis/Postgres for every single user page load so we can populate in very few cases an "already subscribed" UX. That feels like we are doing a lot of processing for very little in return. I think the more performant solution would be to throw up an "already subscribed" message upon submission. That will save us millions of these external db hits a day. 
2) Should we allow the user to enter an email address rather than force them to use their DEV one? This would eliminate the need to send the user email in the async info which just saves some data across the wire. Much less concerned about that than the extra db hit.

Reply:
1. That's totally fair. It's a "cheap" query and I see what you mean by it needs to be populated in very few cases. We explored this avenue as well. It's a far less than ideal UX to have to re-submit to get that confirmation and we felt the cost of doing this wasn't worth that compromise in UX. An alternative route we considered was moving the logic into its own API call from the liquid tag. In that case, it would fire anytime an article with this tag is viewed (only when "needed"). We figured it sticking it in `base_data` was a quick win since 1) `base_data` needs an overall update anyways so we're only slightly if at all, adding to the "tech debt" here 2) it's more consistent with how we're currently doing things, even if sub-optimally and 3) we are going to re-visit this entire thing post-Codeland to open it up more, so we could take what we learn from that to make a more informed decision that's longer-lasting.
2. We reviewed that point extensively when we scoped this project - that's not a road we want to go down right now because it opens up the door for a lot of spam.

Reply:
> We figured it sticking it in base_data was a quick win 

We did this a bunch in the past and it led to me cleaning up a lot of code to help us performance-wise which is why I want to avoid doing it. I think making an API call from the liquid tag sounds like the way to go. I would imagine it would act similarly to how we populate the text on follow buttons.

Reply:
😢  will do.

Reply:
@mstruve I updated it to follow the "follow buttons" approach, thanks for the suggestion! It feels much better now. Let me know if it can be improved further.

@rhymes I restricted the email to `admin` and `super_admin` roles for now. We'll just need to remove my quick and dirty patch before the final release.

Reply:
Thanks for the update! I'm a fan of @mstruve suggestion:

> Should we allow the user to enter an email address rather than force them to use their DEV one? This would eliminate the need to send the user email in the async info which just saves some data across the wire. Much less concerned about that than the extra db hit.

We could actually do this by letting people enter it and, optionally, then validating it server side to make sure it's the correct email. This way we never send back the actual address. Just a thought